modular_skyrat/modules/gladiator/code/modules/mob/living/simple_animal/hostile/megafauna/markedone.dm

```diff
@@ -357,17 +377,50 @@
 			else
 				if(prob(40))
 					INVOKE_ASYNC(src, .proc/bone_knife_throw, target)
-					ranged_cooldown += 3.5 SECONDS
+					ranged_cooldown += 2 SECONDS
 				else
 					INVOKE_ASYNC(src, .proc/teleport, target)
 					ranged_cooldown += 3 SECONDS
 		if(3)
 			if(prob(35))
 				INVOKE_ASYNC(src, .proc/bone_knife_throw, target)
-				ranged_cooldown += 3 SECONDS
+				ranged_cooldown += 1 SECONDS
 			else
 				INVOKE_ASYNC(src, .proc/teleport, target)
 				ranged_cooldown += 2 SECONDS
+		if(4)
+			if(prob(40))
+				INVOKE_ASYNC(src, .proc/swordslam)
+				ranged_cooldown += 3 SECONDS
+			else
+				INVOKE_ASYNC(src, .proc/charge, target, 21)
+				ranged_cooldown += 2 SECONDS
+				
+
+/proc/ground_pound(atom/source, range, delay, throw_range)
+	var/turf/orgin = get_turf(source)
+	if(!orgin)
+		return
+	var/list/all_turfs = RANGE_TURFS(range, orgin)
+	for(var/i = 0 to range)
+		playsound(orgin,'sound/effects/bamf.ogg', 600, TRUE, 10)
+		for(var/turf/stomp_turf in all_turfs)
+			if(get_dist(orgin, stomp_turf) > i)
+				continue
+			new /obj/effect/temp_visual/small_smoke/halfsecond(stomp_turf)
+			for(var/mob/living/getfucked in stomp_turf)
+				if(getfucked == source || getfucked.throwing)
+					continue
+				to_chat(getfucked, span_userdanger("[source]'s ground slam shockwave sends you flying!"))
+				var/turf/thrownat = get_ranged_target_turf_direct(source, getfucked, throw_range, rand(-10, 10))
+				getfucked.throw_at(thrownat, 8, 2, null, TRUE, force = MOVE_FORCE_OVERPOWERING, gentle = TRUE)
+				getfucked.apply_damage(20, BRUTE, wound_bonus=CANT_WOUND)
+				shake_camera(getfucked, 2, 1)
+			all_turfs -= stomp_turf
+		sleep(delay)
+		
+/mob/living/simple_animal/hostile/megafauna/gladiator/proc/swordslam()
```

Comment:
Document this proc, and all the other new procs you've created, to give context as to why they're needed, and what they do.
```suggestion
/mob/living/simple_animal/hostile/megafauna/gladiator/proc/sword_slam()
```